src/OpenTelemetry.Instrumentation.Hangfire/Implementation/HangfireInstrumentationJobFilterAttribute.cs

```diff
@@ -30,8 +34,16 @@ public void OnPerforming(PerformingContext performingContext)
                 return;
             }
 
+            var activityContextData = performingContext.GetJobParameter<Dictionary<string, string>>(HangfireInstrumentationConstants.ActivityContextKey);
+            ActivityContext parentContext = default;
+            if (activityContextData is not null)
+            {
+                var propagationContext = Propagators.DefaultTextMapPropagator.Extract(new PropagationContext(default, Baggage.Current), activityContextData, ExtractActivityProperties);
```

Comment:
Not sure if this is what you intended, but instantiating the context with `Baggage.Current` may prevent baggage from being extracted.

See:
https://github.com/open-telemetry/opentelemetry-dotnet/blob/34efdbf4488afcaa6ade2ae74e4633e10c199514/src/OpenTelemetry.Api/Context/Propagation/BaggagePropagator.cs#L43-L49

You might want to consider something like the following
https://github.com/open-telemetry/opentelemetry-dotnet/blob/34efdbf4488afcaa6ade2ae74e4633e10c199514/examples/MicroserviceExample/Utils/Messaging/MessageReceiver.cs#L58-L62

In the example above, we set Baggage.Current to the extracted baggage. However, if it was your intent to preserve any existing values that may be in baggage, you might need to manually add the extracted baggage to Baggage.Current. I'm not familiar with the flow of Hangfire jobs, so I don't know which of the two options would be more correct.

Reply:
Nice catch! I had no idea...

Fixed it :)